src/test/spec/unified_runner/test_runner.rs

```diff
@@ -60,58 +64,74 @@ const SKIPPED_OPERATIONS: &[&str] = &[
     "watch",
 ];
 
-pub type EntityMap = HashMap<String, Entity>;
+pub(crate) type EntityMap = HashMap<String, Entity>;
 
-pub struct TestRunner {
-    pub internal_client: TestClient,
-    pub entities: EntityMap,
-    pub fail_point_guards: Vec<FailPointGuard>,
+#[derive(Clone)]
+pub(crate) struct TestRunner {
+    pub(crate) internal_client: TestClient,
+    pub(crate) entities: Arc<RwLock<EntityMap>>,
+    pub(crate) fail_point_guards: Arc<RwLock<Vec<FailPointGuard>>>,
 }
 
 impl TestRunner {
-    pub async fn new() -> Self {
+    pub(crate) async fn new() -> Self {
         Self {
             internal_client: TestClient::new().await,
-            entities: HashMap::new(),
-            fail_point_guards: Vec::new(),
+            entities: Default::default(),
+            fail_point_guards: Default::default(),
         }
     }
 
-    pub async fn new_with_connection_string(connection_string: &str) -> Self {
+    pub(crate) async fn new_with_connection_string(connection_string: &str) -> Self {
         #[cfg(all(not(feature = "sync"), not(feature = "tokio-sync")))]
         let options = ClientOptions::parse(connection_string).await.unwrap();
         #[cfg(any(feature = "sync", feature = "tokio-sync"))]
         let options = ClientOptions::parse(connection_string).unwrap();
         Self {
             internal_client: TestClient::with_options(Some(options)).await,
-            entities: HashMap::new(),
-            fail_point_guards: Vec::new(),
+            entities: Arc::new(RwLock::new(EntityMap::new())),
+            fail_point_guards: Arc::new(RwLock::new(Vec::new())),
         }
     }
 
-    pub async fn run_test(&mut self, test_file: TestFile, pred: impl Fn(&TestCase) -> bool) {
+    pub(crate) async fn run_test(
+        &self,
```

Comment:
it seems like we are always creating two test runners now? `run_unified_format_test_filtered` and the workload executor both create them and call this method on them, but then this method creates its own